### PR TITLE
fix(swc): temporarily disable bytecheck

### PR DESCRIPTION
**Pull request by kwonoj**
**Description:**

Reverts https://github.com/swc-project/swc/pull/5062 partially to amend https://github.com/swc-project/swc/issues/5406 as a workaround.

**This is not a correct fix**, since issue itself is quite tricky it requires time to investigate & create some reliable repro steps - meanwhile this PR unblocks existing users. This means potential delay to AST schema version support.

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

**Comment by kwonoj** [comment]
This requires to release new `@swc/core` also.

File: crates/swc_atoms/src/lib.rs [hunk 1]
```diff
@@ -54,7 +54,6 @@ include!(concat!(env!("OUT_DIR"), "/js_word.rs"));
     feature = "rkyv",
     derive(rkyv::Archive, rkyv::Serialize, rkyv::Deserialize)
 )]
-#[cfg_attr(feature = "rkyv", archive_attr(repr(C), derive(bytecheck::CheckBytes)))]
 pub struct Atom(#[cfg_attr(feature = "rkyv", with(crate::EncodeAtom))] Arc<str>);
```
Comment (kdy1) [inline crates/swc_atoms/src/lib.rs:57]: I think we can remove `crate::EncodeAtom`?

Comment (kdy1) [reply crates/swc_atoms/src/lib.rs:57]: Nvm, we can do it later.

Comment (kwonoj) [reply crates/swc_atoms/src/lib.rs:57]: If we're going to reintroduce bytecheck we'll still need it, is that correct? If that's the case I'd limit this revert in minimal, single scoped vhange so we can bring it back easily and also bisect easily later if needed.

**Review by kdy1 — APPROVED** [review]
Thanks!


---



swc-bump:
 - swc_atoms --breaking

File: crates/swc_atoms/src/lib.rs (hunk 1 above)
Comment (kdy1) [reply crates/swc_atoms/src/lib.rs:57]: No it's workaround for https://github.com/rkyv/rkyv/issues/288

**Pull request merged**